### PR TITLE
Log if image is referenced by tag

### PR DESCRIPTION
**Pull request by AdamKorcz**
# Changes
Adds a logging statement if the image is referenced by tag to inform the user of the risk.

/kind enhancement

**Release Note**

```release-note
Log if image is referenced by tag in s2i builder.
```

### CODE REVIEW

File: pkg/builders/s2i/builder.go [hunk 1]
```diff
@@ -356,6 +356,9 @@ func s2iScriptURL(ctx context.Context, cli DockerClient, image string) (string,
 			if err != nil {
 				return "", fmt.Errorf("cannot parse image name: %w", err)
 			}
+			if _, ok := ref.(name.Tag); ok {
```
Comment (matejvasek) [inline pkg/builders/s2i/builder.go:359]: Could we ignore this for default (Red Hat) builder images?
It's not good look if there is warning in default configuration.

**Comment by matejvasek** [comment]
PTAL @jrangelramos @lkingland @gauron99

File: pkg/builders/s2i/builder.go (hunk 1 above)
Comment (AdamKorcz) [reply pkg/builders/s2i/builder.go]: Yes, I can add this check today.

**Comment by matejvasek** [comment]
@ReToCode @lance please override the `Check Source (ubuntu-latest)` error.

File: pkg/builders/s2i/builder.go (hunk 1 above)
Comment (matejvasek) [reply pkg/builders/s2i/builder.go]: I already added it.

**Pull request merged**